diff --git a/globalredirect.module b/globalredirect.module
index 862913f..6a50f5a 100644
--- a/globalredirect.module
+++ b/globalredirect.module
@@ -43,6 +43,12 @@ function globalredirect_init() {
     }
   }
 
+  // Get the query string parameters. If none set, set to NULL
+  $query_string = drupal_get_query_parameters();
+  if (empty($query_string)) {
+    $query_string = NULL;
+  }
+
   // Store the destination from the $_GET as, if we leave it in, drupal_goto()
   // will go to that instead.
   if (isset($_GET['destination'])) {
@@ -50,12 +56,6 @@ function globalredirect_init() {
     unset($_GET['destination']);
   }
 
-  // Get the query string parameters. If none set, set to NULL
-  $query_string = drupal_get_query_parameters();
-  if (empty($query_string)) {
-    $query_string = NULL;
-  }
-
   // Establish the language prefix that should be used, ie. the one that
   // drupal_goto() would use
   $options = array(
